### PR TITLE
Add Burmese dictionary for test

### PR DESCRIPTION
**Pull request by makotokato**
When using Burmese dictionary from ICU4C simply, dictionary segmenter cannot
handle correct segment. We have to consider Extend and SpacingMark of
grapheme segmenter.

### CODE REVIEW

**Comment by makotokato** [comment]
@sffc could you review this?

File: experimental/segmenter/src/dictionary.rs [hunk 1]
```diff
@@ -9,6 +9,14 @@ use core::str::CharIndices;
 use icu_char16trie::char16trie::{Char16Trie, TrieResult};
 use icu_provider::prelude::*;
 
+// SpacingMark and Extend for Myanmar shouldn't be first character of segment.
+//
+// TODO:
```
Comment (sffc) [inline experimental/segmenter/src/dictionary.rs:14]: Nit: Please make sure an issue is filed to track this

File: experimental/segmenter/src/dictionary.rs [hunk 2]
```diff
@@ -53,6 +61,17 @@ impl<'l, 's, Y: DictionaryType<'l, 's> + ?Sized> Iterator for DictionaryBreakIte
                     return Some(next.0 + Y::char_len(next.1));
                 }
                 TrieResult::Intermediate(_) => {
+                    {
+                        // If next character doesn't allow for grapheme, we don't recognize this word as intermediate state.
+                        let tmp = self.iter.clone();
+                        if let Some(tmp_next) = self.iter.next() {
```
Comment (sffc) [inline experimental/segmenter/src/dictionary.rs:67]: Suggestion: Can you use `.peek()` somehow?

**Review by sffc — APPROVED** [review]
Seems plausible. I have some minor comments.

**Review by aethanyc — APPROVED** [review]
Looks fine, and please address Shane's comments. Thanks!

File: experimental/segmenter/src/dictionary.rs (hunk 2 above)
Comment (makotokato) [reply experimental/segmenter/src/dictionary.rs:67]: I will update to use `Peekable` instead.

File: experimental/segmenter/src/dictionary.rs (hunk 1 above)
Comment (makotokato) [reply experimental/segmenter/src/dictionary.rs:14]: Yes.

**Pull request merged**